src/Microsoft.AspNetCore.Build/targets/Compile.targets

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <!-- Put targets attached to the 'Compile' phase of the Standard Lifecycle here. -->
+
+    <Target Name="DotNetBuild" BeforeTargets="Compile">
+        <ItemGroup>
+            <_ToBuild Include="@(Projects)" Condition="'$(BuildSourceOnly)' != 'true' Or '%(Projects.ProjectGroup)' == 'src'" />
+        </ItemGroup>
+
+        <Delete Files="%(_ToBuild.GeneratedBuildInfoFile)" Condition="Exists(%(_ToBuild.GeneratedBuildInfoFile))" />
+        <WriteLinesToFile
```

Comment:
Writing the file took 15ms, and collecting it took 400ms. I think we should try to embed it every single time we make a build (it's extremely useful) and those are pretty minimal parts of our process (especially since they can be run in parallel when I, eventually, enable parallel build)